### PR TITLE
feat(util/package-rules): allow glob pattens in match{Current,New}Value

### PR DESCRIPTION
**Pull request by amezin**
## Changes

Allow glob patterns in `packageRules.match{Current,New}Value`

## Context

- #29167

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -2749,8 +2749,20 @@ Use this field to restrict rules to a particular datasource. e.g.
...
-For example, the following enforces that only `1.*` versions will be used:
+`matchCurrentValue` supports Regular Expressions and glob patterns. For example, the following enforces that only `1.*` versions will be used:
+
+```json
+{
+  "packageRules": [
+    {
+      "matchPackagePatterns": ["io.github.resilience4j"],
+      "matchCurrentValue": "1.*"
+    }
```
Comment (rarkins) [inline docs/usage/configuration-options.md:2760]: Add a non match argument to this rule so it's valid, eg automerge=true

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/validation.spec.ts [hunk 2]
```diff
@@ -163,10 +163,15 @@ describe('config/validation', () => {
...
+          {
+            matchPackageNames: ['bad'],
+            matchNewValue: '/^2(/',
+            enabled: true,
+          },
         ],
       };
       const { errors } = await configValidation.validateConfig('repo', config);
-      expect(errors).toHaveLength(2);
+      expect(errors).toHaveLength(1);
```
Comment (rarkins) [inline lib/config/validation.spec.ts:174]: Why did it go down?

Comment (amezin) [reply lib/config/validation.spec.ts:174]: All previous "invalid" - non-regex - values became valid

**Review by viceice — COMMENTED** [review]
i think this should target v38 because of probably breaking behavior

**Comment by rarkins** [comment]
@viceice I think this is "adding a feature" (glob and exact match) while existing functionality will remain the same, or have I missed something?

**Comment by viceice** [comment]
ok

**Review by rarkins — APPROVED** [review]

**Pull request merged**